### PR TITLE
Implemented transaction storage

### LINKED ISSUE #150: Implement Storage for Transactions

### PR DESCRIPTION
**Pull request by ElginL**
Currently lacking storage for the `DateTime` field which I will be add after the Transaction model is updated with the DateTime field (I will open a separate pr for that).

### CODE REVIEW

File: src/test/java/seedu/address/storage/JsonAdaptedTransactionTest.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    }
+
+    @Test
+    public void toModelType_invalidQuantity_throwsIllegalValueException() {
+        JsonAdaptedTransaction transaction =
+                new JsonAdaptedTransaction(VALID_GOODS, VALID_PRICE, INVALID_QUANTITY, TransactionType.SELL);
+        String expectedMessage = Quantity.MESSAGE_CONSTRAINTS;
+        assertThrows(IllegalValueException.class, expectedMessage, transaction::toModelType);
+    }
+
```
Comment (Jo3LW) [inline src/test/java/seedu/address/storage/JsonAdaptedTransactionTest.java:74]: Great code!

**Review by Jo3LW — APPROVED** [review]
LGTM! Wow great job.

File: src/main/java/seedu/address/model/ReadOnlyClient.java [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 import javafx.collections.ObservableList;
 import seedu.address.model.company.Company;
+import seedu.address.model.transaction.Transaction;
```
Comment (quannam0124) [inline src/main/java/seedu/address/model/ReadOnlyClient.java:5]: Very readable code!

**Review by quannam0124 — APPROVED** [review]
LGTM! Good job Elgin!

**Pull request merged**